website/content/docs/auth/oidc/index.mdx

```diff
@@ -0,0 +1,177 @@
+
+- `mount` (default: "oidc")
+- `listenaddress` (default: "localhost")
+- `port` (default: 8250)
+- `callbackhost` (default: "localhost")
+- `callbackmethod` (default: "http")
+- `callbackport` (default: value set for `port`). This value is used in the `redirect_uri`, whereas
+  `port` is the localhost port that the listener is using. These two may be different in advanced setups.
+
+## Configuration
+
+The OIDC authentication flow has been successfully tested with a number of providers. A full
+guide to configuring OAuth/OIDC applications is beyond the scope of Vault documentation, but a
+collection of provider configuration steps has been collected to help get started:
+[OIDC Provider Setup](/vault/docs/auth/oidc/oidc-providers)
+
+### Redirect URIs
+
+An important part of OIDC role configuration is properly setting redirect URIs. This must be
+done both in Vault and with the OIDC provider, and these configurations must align. The
+redirect URIs are specified for a role with the `allowed_redirect_uris` parameter. There are
+different redirect URIs to configure the Vault UI and CLI flows, so one or both will need to
+be set up depending on the installation.
+
+**CLI**
+
+If you plan to support authentication via `vault login -method=oidc`, a localhost redirect URI
+must be set. This can usually be: `http://localhost:8250/oidc/callback`. Logins via the CLI may
+specify a different host and/or listening port if needed, and a URI with this host/port must match one
+of the configured redirected URIs. These same "localhost" URIs must be added to the provider as well.
+
+**Vault UI**
+
+Logging in via the Vault UI requires a redirect URI of the form:
+
+`https://{host:port}/ui/vault/auth/{path}/oidc/callback`
+
+The "host:port" must be correct for the Vault server, and "path" must match the path the OIDC
+backend is mounted at.
+
+If the [oidc_response_mode](/vault/api-docs/auth/jwt#oidc_response_mode) is set to `form_post`, then
+logging in via the Vault UI requires a redirect URI of the form:
+
+`https://{host:port}/v1/auth/{path}/oidc/callback`
+
+Prior to Vault 1.6, if [namespaces](/vault/docs/enterprise/namespaces) are in use,
+they must be added as query parameters, for example:
+
+`https://vault.example.com:8200/ui/vault/auth/oidc/oidc/callback?namespace=my_ns`
+
+For Vault 1.6+, it is no longer necessary to add the namespace as a query
+parameter in the redirect URI, if
+[`namespace_in_state`](/vault/api-docs/auth/jwt#namespace_in_state) is set to `true`,
+which is the default for new configs.
+
+### OIDC configuration troubleshooting
+
+This amount of configuration required for OIDC is relatively small, but it can be tricky to debug
+why things aren't working. Some tips for setting up OIDC:
+
+- If a role parameter (e.g. `bound_claims`) requires a map value, it can't be set individually using
+  the Vault CLI. In these cases the best approach is to write the entire configuration as a single
+  JSON object:
+
+```text
```

Comment:
Noticed inconsistent use of `text` and `shell-session`. Also not sure what the recommended format is here?

Reply:
@schavis - Also curious if you have a preference for this? (thank you 🙂)

Reply:
I believe the convention is: if the code block includes shell commands or output, use `shell-session`. If it's just plain text, use `text`